### PR TITLE
Clean up errors

### PR DESCRIPTION
**Pull request by aerosol**
This PR enumerates all Clickhouse error codes to be included in the exception struct. It also unifies the way Clickhousex erros are created and wrapped so that further filtering/aggregation (e.g. Sentry fingerprinting) is possible. The Response module was extracted to its own file by the way.

### CODE REVIEW

File: lib/clickhousex/error.ex [hunk 1]
```diff
@@ -665,7 +665,7 @@ defmodule Clickhousex.Error do
   defp get_code(message) do
     case Regex.scan(~r/^Code: (\d+)/i, message) do
```
Comment (ruslandoga) [inline lib/clickhousex/error.ex:666]: Just FYI: we can get the code from `x-clickhouse-exception-code` header.

Comment (ruslandoga) [reply lib/clickhousex/error.ex:666]:
```console
> curl 'http://localhost:8123/?query=select' -v
*   Trying 127.0.0.1:8123...
* Connected to localhost (127.0.0.1) port 8123 (#0)
> GET /?query=select HTTP/1.1
> Host: localhost:8123
> User-Agent: curl/7.84.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 400 Bad Request
< Date: Tue, 10 Jan 2023 14:18:10 GMT
< Connection: Keep-Alive
< Content-Type: text/plain; charset=UTF-8
< X-ClickHouse-Server-Display-Name: mac3.local
< Transfer-Encoding: chunked
< X-ClickHouse-Exception-Code: 62
< Keep-Alive: timeout=10
< X-ClickHouse-Summary: {"read_rows":"0","read_bytes":"0","written_rows":"0","written_bytes":"0","total_rows_to_read":"0","result_rows":"0","result_bytes":"0"}
<
Code: 62. DB::Exception: Syntax error: failed at position 8 (end of query): . Expected one of: ALL, DISTINCT ON, DISTINCT, TOP, not empty list of expressions, list of expressions, list of elements, expression with optional alias, element of expression with optional alias, lambda expression. (SYNTAX_ERROR) (version 22.10.1.1175 (official build))
* Connection #0 to host localhost left intact
```

Comment (ruslandoga) [reply lib/clickhousex/error.ex:666]: I guess the error message format is unlikely to change, but the header is probably even unlikli-er to change :)

File: lib/clickhousex/error.ex [hunk 2]
```diff
@@ -36,16 +669,7 @@ defmodule Clickhousex.Error do
...
-  defp translate("81"), do: :database_does_not_exists
-  defp translate("82"), do: :database_already_exists
-  defp translate("28000"), do: :invalid_authorization
-  defp translate("08" <> _), do: :connection_exception
-  defp translate(code), do: code
-
-  defp get_constraint_violations(_reason) do
-    []
+  for {code, error} <- @clickhouse_codes do
+    defp translate(unquote(code)), do: unquote(String.to_atom(String.downcase(error)))
```
Comment (ruslandoga) [inline lib/clickhousex/error.ex:673]: Why do we need to know the translation? For fingerprinting, the "integer" code should be enough, and then there is the translation and full error message in `message`.

Comment (aerosol) [reply lib/clickhousex/error.ex:673]: You're right, I'll remove it then

File: lib/clickhousex/error.ex (hunk 1 above)
Comment (aerosol) [reply lib/clickhousex/error.ex:666]: Ah cool, I'm not sure it makes sense to work on header parsing for that with a completely new client around the corner. WYT?

**Review by ruslandoga — APPROVED** [review]

File: lib/clickhousex/error.ex (hunk 2 above)
Comment (aerosol) [reply lib/clickhousex/error.ex:673]: @ruslandoga turns out they're useful for tests

```
assert {:error, %{code: :table_already_exists}} ..
```

is better than

```
assert {:error, %{code: "67"}} ..
```

and certainly better than asserting random error text clickhouse returns. WYT?

Comment (aerosol) [reply lib/clickhousex/error.ex:673]: The old codes were wrong, so at least that's some improvement haha

Comment (ruslandoga) [reply lib/clickhousex/error.ex:673]: Yeah, the new asserts are definitely better :) We'd need to keep an eye on ClickHouse adding new errors then. Or maybe not, we can update the list only after we encounter an "unknown" error code.

**Pull request merged**